### ISSUE TITLE
KVM: TestStartStopWithPreload 

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#7184]
as seen here:

https://storage.googleapis.com/minikube-builds/logs/7165/ba4e134/KVM_Linux.html#fail_TestStartStopWithPreload

**Comment by medyagh** [comment]
CC: @priyawadhwa

**Comment by medyagh** [comment]
as seen here too https://storage.googleapis.com/minikube-builds/logs/7202/0478f2d/KVM_Linux.html#fail_TestStartStopWithPreload

**Comment by priyawadhwa** [comment]
This seems to be the preload error:

```
I0324 13:55:49.458798    5115 docker.go:368] k8s.gcr.io/kube-proxy:v1.17.3 wasn't preloaded
I0324 13:55:49.458879    5115 ssh_runner.go:101] Run: sudo cat /var/lib/docker/image/overlay2/repositories.json
I0324 13:55:49.477302    5115 ssh_runner.go:101] Run: which lz4
W0324 13:55:49.483958    5115 start.go:205] Docker preload failed: getting file asset: Error opening file asset: /home/jenkins/minikube-integration/linux-amd64-kvm2-7202-30594-0478f2d2048e7b82627351c9afe530156f0843b9/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v2-v1.17.3-docker-overlay2-amd64.tar.lz4: open /home/jenkins/minikube-integration/linux-amd64-kvm2-7202-30594-0478f2d2048e7b82627351c9afe530156f0843b9/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v2-v1.17.3-docker-overlay2-amd64.tar.lz4: no such file or directory, falling back to caching images
```

so something is broken on restart

and minikube actually fails because of a coredns error:

```
error execution phase addon/coredns: unable to create a new DNS service: Post https://192.168.39.120:8443/api/v1/namespaces/kube-system/services?timeout=10s: dial tcp 192.168.39.120:8443: connect: connection refused
To see the stack trace of this error execute with --v=5 or higher
* 
X Error starting cluster: addon phase cmd:"/bin/bash -c \"sudo env PATH=/var/lib/minikube/binaries/v1.17.3:$PATH kubeadm init phase addon all --config /var/tmp/minikube/kubeadm.yaml\"": /bin/bash -c "sudo env PATH=/var/lib/minikube/binaries/v1.17.3:$PATH kubeadm init phase addon all --config /var/tmp/minikube/kubeadm.yaml": Process exited with status 1
stdout:
[addons]: Migrating CoreDNS Corefile
stderr:
W0324 13:56:17.204052    5530 validation.go:28] Cannot validate kube-proxy config - no validator is available
W0324 13:56:17.204322    5530 validation.go:28] Cannot validate kubelet config - no validator is available
error execution phase addon/coredns: unable to create a new DNS service: Post https://192.168.39.120:8443/api/v1/namespaces/kube-system/services?timeout=10s: dial tcp 192.168.39.120:8443: connect: connection refused
To see the stack trace of this error execute with --v=5 or higher
* 
* minikube is exiting due to an error. If the above message is not useful, open an issue:
- https://github.com/kubernetes/minikube/issues/new/choose
```

**Comment by priyawadhwa** [comment]
Can be closed by #7210
